app/models/concerns/user_encrypted_attribute_overrides.rb

```diff
@@ -0,0 +1,128 @@
+module UserEncryptedAttributeOverrides
+  extend ActiveSupport::Concern
+
+  attr_accessor :attribute_user_access_key
+
+  # override some Devise methods to support our use of encrypted_email
+
+  class_methods do
+    def find_first_by_auth_conditions(tainted_conditions, opts = {})
+      if tainted_conditions[:email].present?
+        opts[:email_fingerprint] = create_fingerprint(tainted_conditions.delete(:email))
+      end
+      to_adapter.find_first(devise_parameter_filter.filter(tainted_conditions).merge(opts))
+    end
+
+    def find_with_email(email)
+      email_fingerprint = create_fingerprint(email)
+      find_by(email_fingerprint: email_fingerprint)
+    end
+
+    def create_fingerprint(email)
+      Pii::Fingerprinter.fingerprint(email.downcase)
+    end
+  end
+
+  # Override Devise::Models::Confirmable in order to
```

Comment:
Devise model is pretty baked in, even if our controllers don't need to inherit from Devise. I don't have feelings one way or another about keeping it.